pkgs/build-support/setup-hooks/drop-build-tree.sh

```diff
@@ -0,0 +1,5 @@
+postPhases+=" removeBuildDir"
+
+removeBuildDir() {
+    rm -rf $out/.build
```

Comment:
It feels that this should be part of `keepBuildTree`.

Reply:
I guess it should not...? `dropBuildTree` means to cancel the effect of  `keepBuildTree` on packages not depending on `$NIX_BUILD_TOP`.

Reply:
There is an implicit contract on the path `$out/.build` between the two hooks (and there's no reference to `keepBuildTree` treewide), so IMO they'd better be located in the same `.sh`.